DataFormats/ForwardDetId/interface/MTDChannelIdentifier.h

```diff
@@ -0,0 +1,85 @@
+#ifndef DATAFORMATS_MTDCHANMNELIDENTIFIER_H
+#define DATAFORMATS_MTDCHANMNELIDENTIFIER_H
+
+#include <utility>
+#include "FWCore/Utilities/interface/GCC11Compatibility.h"
+
+class MTDChannelIdentifier{
+ public:
+  
+  typedef unsigned int PackedDigiType;
+  typedef unsigned int ChannelType;
+  
+  static std::pair<int,int> channelToPixel( int ch) {
+    int row = ( ch >> thePacking.column_width) & thePacking.row_mask;
+    int col = ch & thePacking.column_mask;
+    return std::pair<int,int>(row,col);
+  }
+  
+  static int pixelToChannel( int row, int col) {
+    return (row << thePacking.column_width) | col;
+  }
+  
+  /**
+   * Pack the pixel information to use less memory
+   */
+
+  class Packing {
+  public:
+    
+    // Constructor: pre-computes masks and shifts from field widths
+    // gcc4.8: sorry, unimplemented: use of the value of the object being constructed in a constant expression
```

Comment:
do we really need to worry about compatibility with C++0x and gcc4.8? In 10_4_X we use gcc700 and C++17...

Reply:
Nah, I'll get rid of it.